bms-monitor.js

```diff
@@ -1,18 +1,30 @@
-const {expect} = require('chai');
+const {TemperatureRange, SOCRange, ChargeRateRange}=require("./inputParameterRanges.js")
 
-function batteryIsOk(temperature, soc, charge_rate) {
-    if (temperature < 0 || temperature > 45) {
-        console.log('Temperature is out of range!');
+//checks whether inputParameters are in range or not and specifies the type of breach that is low or high
+function checkBreach(inputParameter, value, min, max) {
+    if (value < min) {
+        console.log(`!!! ${inputParameter} is low that is ${value} - Not in range!!!`);
         return false;
-    } else if (soc < 20 || soc > 80) {
-        console.log('State of Charge is out of range!')
-        return false;
-    } else if (charge_rate > 0.8) {
-        console.log('Charge rate is out of range!');
+    } else if (value > max) {
+        console.log(`!!! ${inputParameter} is high that is ${value} - Not in range!!!`);
         return false;
     }
-    return true;
+        console.log( `${inputParameter} is within the valid range that is : ${value}`) ;
+        return true;    
 }
 
-expect(batteryIsOk(25, 70, 0.7)).to.be.true;
-expect(batteryIsOk(50, 85, 0)).to.be.false;
+function batteryIsOk(temperature, soc, chargeRate){
+    const temperatureResult=checkBreach( "Temperature",temperature,TemperatureRange.min,TemperatureRange.max);
+    const socResult=checkBreach("soc", soc, SOCRange.min, SOCRange.max);
+    const chargeRateResult=checkBreach("chargeRate", chargeRate, ChargeRateRange.min, ChargeRateRange.max);
+    
+    if(temperatureResult & socResult & chargeRateResult ){

```

Comment:
The complexity of this condition increases by adding more parameters. Can you try changing, to add a parameter without increasing complexity?